scipy/signal/sigtoolsmodule.c

```diff
@@ -1011,6 +1015,15 @@ PyObject *PyArray_OrderFilterND(PyObject *op1, PyObject *op2, int order) {
 
 fail:
 	if (zptr) PyDataMem_FREE(zptr);
+	if (sort_buffer) free(sort_buffer);
```

Comment:
The variables need to be explicitly initialized to NULL in the declarations at the beginning.